src-tauri/src/constructors/init/init_graph.rs

```diff
@@ -47,11 +49,28 @@ pub fn init_graph(
     graph
 }
 
-pub fn add_node_to_graph_if_not_exists(graph: &mut Graph, node_id: u32) {
+pub fn add_node_and_location_to_graph(
+    node_id: u32,
+    graph: &mut Graph,
+    node_loc: Option<&MeshNode>,
+) -> NodeIndex {
     let name: String = node_id.to_string();
     if !graph.contains_node(name.clone()) {
-        graph.add_node(name);
+        let mut node = Node::new(name);
+        if let Some(node_loc) = node_loc {
+            let node_pos = &node_loc.data.position;
+            if let Some(node_pos) = node_pos {
+                node.latitude = node_pos.latitude_i as f64 * LAT_CONVERSION_FACTOR;
+                node.longitude = node_pos.longitude_i as f64 * LON_CONVERSION_FACTOR;
+                node.altitude = node_pos.altitude as f64 * ALT_CONVERSION_FACTOR;
+                node.speed = node_pos.ground_speed as f64 * SPEED_CONVERSION_FACTOR;
+                node.direction = 0.0;
```

Comment:
How would we get/update this information?

Reply:
`set_gps`, updated just now

Reply:
I'm not sure if it clones or actually updates the info, so I'll have to test it